crates/net/network/src/session/active.rs

```diff
@@ -322,9 +335,21 @@ impl ActiveSession {
         for id in timedout {
             warn!(target: "net::session", ?id, remote_peer_id=?self.remote_peer_id, "timed out outgoing request");
             let req = self.inflight_requests.remove(&id).expect("exists; qed");
+            self.update_request_timeout(req.timestamp, req.deadline);
             req.request.send_err_response(RequestError::Timeout);
         }
     }
+
+    /// Updates the request timeout with a request's timestamps
+    fn update_request_timeout(&mut self, sent: Instant, received: Instant) {
```

Comment:
this makes sense to me,

I'd like to see some sanity tests for this, I think we can split this into to parts:

* standalone,testable function that does the computation
* `update_request_timeout` calls this function and updates the time out

Reply:
Done! Should we `#[inline]` it?

Reply:
sg, let's do that!